osidb/serializer.py

```diff
@@ -216,6 +220,42 @@ class Meta:
         fields = ["created_dt", "updated_dt"]
         abstract = True
 
+    def update(self, instance, validated_data):
+        """
+        set instance updated timestamp to the one in the request
+        so the TrackingMixin can later perform the comparison
+
+        it is not actually used when saving the instance to DB
+        as that timestamp is being obtained automatically
+        """
+        instance.updated_dt = make_aware(
+            datetime.strptime(
+                self.context["request"].data.get("updated_dt"), DATETIME_FMT
+            )
+        )
+
+        try:
+            return super().update(instance, validated_data)
+        except DataInconsistencyException as e:
+            raise BadRequest(
+                "Received model contains non-refreshed and outdated data! "
+                "It has been probably edited by someone else in the meantime"
+            ) from e
+
+    def validate(self, data):
+        """
+        validate that the updated_dt timestamp was provided
+        because it is used to detect the mid-air collisions
+        """
+        # the updated_dt field is required but read-only
+        # so it is not present as part of the serialized data
+        updated_dt = self.context["request"].data.get("updated_dt")
+        # check that the instance exists so this is an update
+        if self.instance and updated_dt is None:
+            raise ValidationError({"updated_dt": "Field is required"})
```

Comment:
If we use the approach which I've suggested in the previous comment this might not be necessary as the validation of field will do it anyway.